static/src/javascripts/projects/commercial/modules/brazeBanner.js

```diff
@@ -154,14 +155,25 @@ const getMessageFromBraze = async (apiKey: string, brazeUuid: string): Promise<b
             return;
         }
 
-        appboy.subscribeToInAppMessage((message: InAppMessage) => {
+        let subscriptionId: ?string;
+
+        const callback = (message: InAppMessage) => {
             if (message.extras) {
                 messageConfig = message;
                 resolve(true);
             } else {
                 resolve(false);
             }
-        });
+
+            // Unsubscribe
+            if (appboy && subscriptionId) {
+                appboy.removeSubscription(subscriptionId);
+            }
+        };
+
+        // Keep hold of the subscription ID so that we can unsubscribe in the
+        // callback, ensuring that the callback is only invoked once per page
+        subscriptionId = appboy.subscribeToInAppMessage(callback);
```

Comment:
Can we hoist this above the `canShowPromise` declaration? This is valid because that callback won't be called until after this line has been evaluated, of course. It might help the reader to see that `subscriptionId` exists before it is used, though.

Reply:
No! It is passed the callback as an argument, of course.